### PR TITLE
Release/2.3.0

### PR DESCRIPTION
**Pull request by altaf-ali**

### CODE REVIEW

File: weaver/app.R [hunk 1]
```diff
@@ -457,17 +454,28 @@ server <- function(input, output, session) {
...
+          select(
+            directories,
+            project,
+            directory_path,
+            num_files,
+            size,
+            last_modified,
+            filetypes
+          ),
+          colnames = c("Project", "Path", "Number of Files", "Size (GiB)", "Last Modified (days)", "File Usage (GiB)"),
```
Comment (sb10) [inline weaver/app.R:458]: Is this missing a column name for directories?

File: weaver/db.R [hunk 2]
```diff
@@ -19,46 +19,81 @@
...
-    mutate(is_humgen_yn = ifelse(is_humgen == 1, "Yes", "No"), archived_yn = ifelse(archived == 1, "Yes", "No"))
-
-    return(results)
-
+      quota_use = na_if(round(used_gib * 100/quota_gib, digits = 2), Inf),
+      quota_gib = na_if(quota_gib, 0)
+    ) %>% 
+    rowwise() %>% 
+    mutate(
+      path = fs::path_rel(directory_path, fs::path("/lustre", scratch_disk))
```
Comment (sb10) [inline weaver/db.R]: Is there still a requirement for us to hardcode "/lustre" strings? Any way around this?

**Review by sb10 — COMMENTED** [review]
I don't know this code base, so apologies if my comments aren't valid.

Is there any way to write tests in this system? Or was the only way to validate your changes by manually checking the effects they had on a running website?

**Review by michael-grace — COMMENTED** [review]
Well, it works, which is good. The only issues I've come across are data issues, which either stem from the data entry being bad (which are being fixed), or because of badly organised data on lustre, and there isn't much we can do about that other than get the data organised properly - it isn't faesible to come up with metrics for analysing this stuff when some groups just decide to leave data scattered all over the place. We can discuss this.

Two little things:
- Although _yes_ technically R can do implicit returns of a value defined in a function, I don't think it's particularly clear to remove the explicit `return` statement. But that may just be me, though anyone who is used to _most other languages_ would expect an explicit `return` statement.
- There's a lot of merge commits here - would be good if these were rebased out (they look to only be merge commits done with a fast forward, which GitHub does by default when hitting the `merge with merge commit` button:
<table>
<tr><th>Commit</th><th>Thoughts</th></tr>
<tr><td>`99609b0`</td><td>Keep</td></tr>
<tr><td>`78a3720`</td><td>Keep or Squash</td></tr>
<tr><td>`1d5f768`</td><td>Remove</td></tr>
<tr><td>`455b764`</td><td>Keep</td></tr>
<tr><td>`6ca3b57`</td><td>Remove</td></tr>
<tr><td>`062e08c`</td><td>Keep</td></tr>
<tr><td>`4bf8e00`</td><td>Remove</td></tr>
<tr><td>`c5328a9`</td><td>Remove</td></tr>
</table>

**Comment by altaf-ali** [comment]
> Well, it works, which is good. The only issues I've come across are data issues, which either stem from the data entry being bad (which are being fixed), or because of badly organised data on lustre, and there isn't much we can do about that other than get the data organised properly - it isn't faesible to come up with metrics for analysing this stuff when some groups just decide to leave data scattered all over the place. We can discuss this.
> 
> Two little things:
> 
> * Although _yes_ technically R can do implicit returns of a value defined in a function, I don't think it's particularly clear to remove the explicit `return` statement. But that may just be me, though anyone who is used to _most other languages_ would expect an explicit `return` statement.

Incorrect. I've had this argument with others in the past. The definitive authority on this is of course Hadley who says:

"Returning the last evaluated expression is an absolutely key part of the R language — if you don't understand this you are going to find it hard to understand much code in the wild. Since you have to know this, I think it's much more important to reserve return() to signal early returns for exceptional conditions."

> * There's a lot of merge commits here - would be good if these were rebased out (they look to only be merge commits done with a fast forward, which GitHub does by default when hitting the `merge with merge commit` button:
> 
> Commit	Thoughts
> `99609b0`	Keep
> `78a3720`	Keep or Squash
> `1d5f768`	Remove
> `455b764`	Keep
> `6ca3b57`	Remove
> `062e08c`	Keep
> `4bf8e00`	Remove
> `c5328a9`	Remove

I think if I do a squash and merge to main, it should address this.

**Comment by michael-grace** [comment]
_adds another thing to the list of reasons I think R is an awful language_

**Comment by sb10** [comment]
[get weaver to display base directories](https://trello.com/c/6NKovCDb/16-get-weaver-to-display-base-directories)

**Pull request merged**